pandas/core/dtypes/cast.py

```diff
@@ -398,12 +405,31 @@ def maybe_promote(dtype, fill_value=np.nan):
         pass
     elif is_datetime64tz_dtype(dtype):
         pass
-    elif issubclass(np.dtype(dtype).type, str):
+    elif issubclass(np.dtype(dtype).type, (bytes, str)):
         dtype = np.object_
 
     return dtype, fill_value
 
 
+def _check_lossless_cast(value, dtype: np.dtype) -> bool:
```

Comment:
this is nice, though I think we have several implementation of this (or maybe for the array form) in this file, in the future would be nice to consolidate.